core/pagination/pagination.go

```diff
@@ -0,0 +1,38 @@
+package pagination
+
```

Comment:
Pagination would be a better fit for `pkg` instead of `core` since it does not contain any logic specific to Frontier, and can be used fairly independently.

Reply:
Moved to pkg